src/components/FormCK/FormCK.js

```diff
@@ -0,0 +1,196 @@
+import DefaultLeftCopy from './DefaultLeftCopy';
+import ThankYou from './ThankYou';
+
+const FormCK = ({ copyBefore, submitText, formId, children}) => {
+
+    const { register, errors, handleSubmit } = useForm();
+    const [isSubmitting, setIsSubmitting] = useState(false);
+    const [isSubmitted, setIsSubmitted] = useState(false);
+
+    const data = useStaticQuery(getDefaultFormId);
+    if (!formId) {
+        formId  = data.site.siteMetadata.convertkit.defaultFormId
+    }
+    
+    const onSubmit = async (data) => {
+        setIsSubmitting(true);
+        //Id address is filled then it's spam
+        if (!data.address) {
+            const url = `https://api.convertkit.com/v3/forms/${formId}/subscribe`;
+            var headers = {
+                "Content-Type": 'application/json; charset="utf-8"',
+            }
+            var bodyData = {
+                api_key: process.env.GATSBY_CONVERTKIT_APIKEY,
+                email: data.email,
+                first_name: data.name
+            }
+
+            try {
+                const response = await fetch(url, { method: 'POST', headers, body: JSON.stringify(bodyData)});
+                const json = await response.json();
+                if (response.ok && json?.subscription?.id) {
+                    setIsSubmitted(true);
+                }
+            } catch (error) {
+                console.log("Error", error);
+            }
+        }
+
+        setIsSubmitting(false);
+    }
+
+    return (
+        <FormCkWrapper>
+            {copyBefore}
+            <div className="copy-content">
+                {isSubmitted? (
+                    <Box
+                        px={4}
+                        style={{
+                            textAlign: 'center'
+                        }}>
+                        <ThankYou />
+                    </Box>
+                ) : (
+                    <>
+                    <div className="copy-left">
+                        {children}
+                    </div>
+                    <Box
+                        as="form"
+                        onSubmit={handleSubmit(onSubmit)}>
+                        <Input
+                            id="name"
+                            type="text" 
+                            name="name"
+                            ref={register({ required: true})}
+                            placeholder="Your first name"
+                        />
+                        {errors.name && <span><span role="img" aria-label="danger">⚠️</span> Name is required</span>}
+                        <Input
+                            id="email"
+                            type="email" 
+                            name="email"
+                            ref={register({ 
+                                required: "⚠️ E-mail is required",
+                                pattern: {
+                                    value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
```

Comment:
Did you copy this regex from somewhere? I find they're easy to get wrong

Reply:
I think I copied it from react-hook-form examples.. I will search for a better one, do you have one you use?

Reply:
That's fine then 👍 

Just making sure you didn't handcraft because I've definitely made that mistake before.